### DISCUSSION TITLE
How do I run Whisper on a Raspberry Pi 4B?

### DISCUSSION THREAD

**Discussion by abraha2** [openai/whisper#1304] [Q&A]
Hi, I am a Raspberry Pi user, running the official Raspberry Pi OS.
I want to run whisper on my Raspberry Pi 4B, but when I try to install it via pip and pip3, it errors out, saying there are "Conflicting dependencies."
This is the command I used:
pip3 install openai-whisper
And
pip install openai-whisper
They both errored out the same way.
I have done a little research on how to fix this general error, but none of the fixes provided are for anything near to the error I was getting.
Could you please tell me how I might fix this?
Thanks in advance!

**Comment by jongwook** [comment]
I think it's likely related to pytorch installation. Can you run `import torch` without problem?

You can save a file that contains more detailed logs by running: `pip install openai-whisper --log pip.log`, and the content of `pip.log` should contain more clues.

**Reply by abraha2** [reply]
Well, there we go!
Torch did not pass the import test.
Does this mean that I have to reinstall/fix torch, before whisper will run?

**Reply by jongwook** [reply]
Yes, Whisper will need torch installed, and it appears that the `pip` command isn't doing a good job installing the correct packages for Raspberry Pi. I think I had a better luck with miniconda. [this guide](https://gist.github.com/irdanish11/263ffffacbbee2a4dd8f2949fb9e2b1b) seems to be relevant, although you'll probably need to replace the download URL with:

```
wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-aarch64.sh
```

if you're using the 64-bit version (aarch64) of the OS, and the 32-bit version of miniconda (armv7l) is too old (2015) at this point. You can check your OS's architecture by running:

```
uname -m
```

**Reply by abraha2** [reply]
I am running the 32 bit version, does this mean I can follow the guide step-by-step?

**Reply by hwertz** [reply]
> I am running the 32 bit version, does this mean I can follow the guide step-by-step?

Probably, but the turbo model (default) needs 3.5GB RAM and large model 7.8GB (this is what I observed on an Intel system, running on CPU, but should be very similar on ARM.).  Depending on how that kernel is set up (user/kernel address split) your maximum process size is 2-3GB.  So I think the small model is the largest you can run on there.  Possibly medium but it'd be a tight fit.

**Comment by solarsamuel** [comment]
The easiest for Raspberry Pi 4B is to install the C++ version of whisper (not the python version).

Install video is here: https://www.youtube.com/watch?v=caaKhWcfcCY

**Reply by ghost** [reply]
Hello, can Raspberry Pi 5 run Whisper (Python version)?

**Reply by unitydevelopers-greg** [reply]
yes

**Comment by samair** [comment]
Yes, it works  on raspberry pi 5 without any issues.
